katello/katello-change-hostname

```diff
@@ -0,0 +1,93 @@
+
+STDOUT.puts "Starting the hostname change..."
+
+STDOUT.puts("\nA hostname change requires a manual change in Satellite. \n
+            Please log into UI and rename the default capsule (under infrastructure->capsules) 
+            to the new hostname using both the name and url fields.")
+
+unless HighLine.agree('Have you made this necessary change and want to proceed with the hostname change? [yes/no]')
+  STDOUT.puts "Hostname change aborted, no changes have been made to your system"
+  exit
+end
+
+if ARGV[0] && ARGV.count == 1
+  NEW_HOSTNAME = ARGV[0]
+else
+  STDOUT.puts "USAGE: katello-change-hostname <new-hostname>"
+  exit
+end
+
+# Get the hostname from your system
+OLD_HOSTNAME = Socket.gethostname
+
+RHEL_VERSION = `rpm -q --queryformat '%{VERSION}' redhat-release-server`[0]
+
+STDOUT.puts "stopping services"
+`katello-service stop`
+
+STDOUT.puts "deleting old certs"
+`
+rm -rfv /etc/pki/katello{,.bak}
+rm -rfv /etc/pki/katello-certs-tools{,.bak}
+rm -rfv /etc/candlepin/certs/amqp{,.bak}
+rm -rfv /etc/foreman-proxy/*ssl*
+rm -rfv /etc/foreman/old-certs
+rm -rfv /etc/foreman/*.pem
+rm -rfv /var/lib/puppet/ssl
+rm -rfv /root/ssl-build
+`
+
+if RHEL_VERSION.to_i == 6
+  STDOUT.puts "updating hostname in /etc/sysconfig/network"
+  `sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/sysconfig/network`
+elsif RHEL_VERSION.to_i == 7
+  STDOUT.puts "updating hostname in /etc/hostname"
+  `sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/hostname`
+
+  STDOUT.puts "setting hostname"
+  `hostnamectl set-hostname #{NEW_HOSTNAME}`
+else
+  STDOUT.puts "couldn't find RHEL version: #{RHEL_VERSION}"
+  exit
+end
+
+STDOUT.puts "updating hostname in /etc/hosts"
+`sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/hosts`
+
+STDOUT.puts "checking if hostname was changed"
+hostname = Socket.gethostname
+if hostname != NEW_HOSTNAME
+  STDOUT.puts "the new hostname was not changed, check hostname -f"
+  exit
+end
+
+STDOUT.puts "updating hostname in /etc/foreman-installer/scenarios.d/satellite-answers.yaml"
+`sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/foreman-installer/scenarios.d/satellite-answers.yaml`
+
+STDOUT.puts "re-running the Satellite installer"
+`
+satellite-installer --scenario satellite -v --certs-regenerate-ca=true --certs-regenerate=true \
+  --foreman-foreman-url "https://#{NEW_HOSTNAME}" \
+  --foreman-servername #{NEW_HOSTNAME} \
+  --capsule-qpid-router-broker-addr  #{NEW_HOSTNAME}\
+  --certs-ca-common-name #{NEW_HOSTNAME} \
+  --certs-node-fqdn #{NEW_HOSTNAME} \
+  --capsule-parent-fqdn #{NEW_HOSTNAME}\
+  --foreman-proxy-register-in-foreman true
+`
+
+STDOUT.puts "Restarting services"
+`katello-service restart`
+
+STDOUT.puts "
+Hostname change complete!
+
+If you have a capsule you will need to reregister it with RHSM and then run the following (replacing <capsule-hostname> with your capsule's hostname):
+capsule-certs-generate --capsule-fqdn '<capsule-hostname>' --certs-tar '~/<capsule-hostname>-certs.tar'
```

Comment:
Does this mean the default capsule or ALL external capsules will need to regenerate certificate bundles?